packages/substrate-bindings/src/codecs/blockHeader.ts

```diff
@@ -0,0 +1,36 @@
+import { Bytes, Enum, Struct, Vector, _void, enhanceCodec } from "scale-ts"
+import { Hex } from "./Hex"
+import { compactNumber } from "./compact"
+
+const textEncoder = new TextEncoder()
+const textDecoder = new TextDecoder()
+
+const fourChars = enhanceCodec(
+  Bytes(4),
+  textEncoder.encode.bind(textEncoder),
+  textDecoder.decode.bind(textDecoder),
+)
+
+const diggestVal = Struct({
+  engine: fourChars,
+  payload: Hex(),
+})
+
+const diggest = Enum(
+  {
+    consensus: diggestVal,
+    seal: diggestVal,
+    preRuntime: diggestVal,
+    runtimeUpdated: _void,
+  },
+  [4, 5, 6, 8],
+)
```

Comment:
I would add a comment regarding the `other` variant

https://github.com/paritytech/polkadot-sdk/blob/19de1c96607f80ea1f55584c42e7050df08cb3e2/substrate/primitives/runtime/src/generic/digest.rs#L174-L193

https://github.com/paritytech/polkadot-sdk/blob/19de1c96607f80ea1f55584c42e7050df08cb3e2/substrate/primitives/runtime/src/generic/digest.rs#L201-L207

Reply:
I actually decided to purposely remove it after I properly [reviewed the spec](https://spec.polkadot.network/chap-state#defn-digest).

I mean, we only need a comment if we add it, right? So, I decided not to add it because well, it's not being used, so...